plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/worker/BaselineRules.kt

```diff
@@ -0,0 +1,92 @@
+package org.jlleitschuh.gradle.ktlint.worker
+
+import com.pinterest.ktlint.core.LintError
+import java.lang.reflect.Method
+
+/**
+ * This file contains reflection code for accessing the Baseline API which had a breaking change in 0.46 and again in 0.47
+ * This function is the entry point, which will load the lint errors from the baseline report in a version-neutral way
+ */
+internal fun loadBaselineRules(path: String): Map<String, List<LintError>> {
+    return newBaselineClass?.let { loadNewBaselineRules(path) }
+        ?: baselineClass46?.let { load46BaselineRules(path) }
+        ?: loadOldBaselineRules(path)
+}
```

Comment:
How often is this executed? Once per task? Once per file audited?

Reply:
Once per report generated (same as before). Since multiple reports can be generated, i put the reflection heavy lifting behind a shared lazy property